src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -222,51 +223,242 @@ private bool ScanDesignation(bool permitTuple)
+            var list = _pool.AllocateSeparated<SubpatternElementSyntax>();
+            try
+            {
+                if (this.CurrentToken.Kind != closeKind && this.CurrentToken.Kind != SyntaxKind.SemicolonToken)
+                {
+                    tryAgain:
+
+                    if (this.IsPossibleSubpatternElement() || this.CurrentToken.Kind == SyntaxKind.CommaToken)
+                    {
+                        // first pattern
+                        list.Add(this.ParseSubpatternElement());
 
-            return node;
+                        // additional patterns
+                        while (true)
+                        {
+                            if (this.CurrentToken.Kind == SyntaxKind.CloseParenToken ||
+                                this.CurrentToken.Kind == SyntaxKind.CloseBraceToken ||
+                                this.CurrentToken.Kind == SyntaxKind.SemicolonToken)
+                            {
+                                break;
+                            }
+                            else if (this.CurrentToken.Kind == SyntaxKind.CommaToken || this.IsPossibleSubpatternElement())
+                            {
+                                list.AddSeparator(this.EatToken(SyntaxKind.CommaToken));
+                                list.Add(this.ParseSubpatternElement());
+                                continue;
+                            }
+                            else if (this.SkipBadPatternListTokens(ref openToken, list, SyntaxKind.CommaToken, closeKind) == PostSkipAction.Abort)
+                            {
+                                break;
+                            }
+                        }
+                    }
+                    else if (this.SkipBadPatternListTokens(ref openToken, list, SyntaxKind.IdentifierToken, closeKind) == PostSkipAction.Continue)
+                    {
+                        goto tryAgain;
+                    }
+                }
+
+                closeToken = this.EatToken(closeKind);
+                subPatterns = list.ToList();
+            }
+            finally
+            {
+                _pool.Free(list);
+            }
+        }
+
+        private SubpatternElementSyntax ParseSubpatternElement()
+        {
+            NameColonSyntax nameColon = null;
+            if (this.CurrentToken.Kind == SyntaxKind.IdentifierToken && this.PeekToken(1).Kind == SyntaxKind.ColonToken)
+            {
+                var name = this.ParseIdentifierName();
+                var colon = this.EatToken(SyntaxKind.ColonToken);
+                nameColon = _syntaxFactory.NameColon(name, colon);
+            }
+
+            var pattern = ParsePattern();
+            return this._syntaxFactory.SubpatternElement(nameColon, pattern);
+        }
+
+        private bool IsPossibleSubpatternElement()
+        {
+            return this.IsPossibleExpression() || this.CurrentToken.Kind == SyntaxKind.OpenBraceToken;
```

Comment:
>return this.IsPossibleExpression() || this.CurrentToken.Kind == SyntaxKind.OpenBraceToken; [](start = 12, length = 90)

This logic requires some explanation. As part of the review and as a comment in code.

Reply:
A pattern either start with something that could start an expression, or it starts with an open brace token. That's what the code says, and I'm not sure what else can be said about it.

---
In reply to: [151816738](https://github.com/dotnet/roslyn/pull/23208#discussion_r151816738) [](ancestors = 151816738)

Reply:
I assume that is true for NameColon synatx. Explicitly stating all that in the comment will be sufficient.

---
In reply to: [151818262](https://github.com/dotnet/roslyn/pull/23208#discussion_r151818262) [](ancestors = 151818262,151816738)

Reply:
`IsPossible` methods only look at one token. Recorded in https://github.com/dotnet/roslyn/issues/23222 to say that in a comment.